tox.ini

```diff
@@ -44,8 +44,8 @@ commands =
     # Clean up from previous tests
     python {toxinidir}/util.py clean-tests
     # Run tests
-    pytest -vv -rs --cov=sqlfluff --cov-report=xml -m "not dbt_connection_failure" {posargs: {toxinidir}/test}
-    dbt{017,018,019,020,021,100}: pytest -vv -rs --cov=sqlfluff --cov-report=xml -m "dbt_connection_failure" {posargs: test} # Run the connection failure test in a separate invocation to reset the database connection
```

Comment:
These are no longer necessary, just resetting

Reply:
I think these might be necessary, as otherwise it uses the DB connection already established for the previous linting tests.

Reply:
Oh maybe not if you've found the alternative reason for it failing above. Ignore me.

Reply:
@barrywhart I've combined these two lines and now the mocked dbt function isn't happy. That makes me think that the mocking isn't currently removing the need for re-establishing the dbt connections?

Reply:
but actually these test pass perfectly fine for me locally in the docker-compose environement 🤔

Reply:
What do you mean by this?
>now the mocked dbt function isn't happy

Reply:
I see that the dbt 1.0 tests are passing, but the older dbt versions are failing. That makes sense in terms of the mock, because the mock only comes into play for dbt 1.0. We can do one of two things:
* Only run the test if dbt version >= 1.0
*  Update the test to also mock & fail on older dbt versions

I suggest the former, because all this (the test, the PR, everything) is new for dbt 1.0 anyway.

Reply:
But why does it work in the original PR without these additional changes?

Reply:
ah good shout thanks 😄

Reply:
Ok, here's what I did (first 3 lines are new):
```
@pytest.mark.skipif(
    DBT_VERSION_TUPLE < (1, 0), reason="mocks a function that's only used in dbt >= 1.0"
)
@mock.patch("dbt.adapters.postgres.impl.PostgresAdapter.set_relations_cache")
@pytest.mark.dbt_connection_failure
def test__templater_dbt_handle_database_connection_failure(
    set_relations_cache, project_dir, dbt_templater  # noqa: F811
):
```

Reply:
@tunetheweb: IIUC, in dbt < 1.0, a database connection was not required. And the version of the test in `main` **only** checks for (i.e. enforces) a connection failure if dbt version >= 1.0.

https://github.com/sqlfluff/sqlfluff/blob/main/plugins/sqlfluff-templater-dbt/test/templater_test.py#L352L360

Reply:
@jpy-git: Note that I pushed the test change to Niall's branch, not yours. So your PR branch will need updating in order to pick up this change.